presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java

```diff
@@ -429,7 +432,7 @@ public ReadContext init(
             ImmutableList.Builder<parquet.schema.Type> fields = ImmutableList.builder();
             for (HiveColumnHandle column : columns) {
                 if (!column.isPartitionKey() && column.getHiveColumnIndex() < messageType.getFieldCount()) {
-                    fields.add(messageType.getType(column.getName()));
+                    fields.add(getParquetType(configuration, column, messageType));
```

Comment:
@dain I just tested. This is not working when the metastore has column reordering, since the column index is no longer reliable. 
I am thinking if the metastore has column reordering(and no column rename), should use column name access, if the metastore has column reordering and column rename, there is no way to get it right.
Originally, there are 2 cases:
#1 when metastore has column reorder(or column delete), use column name access
#2 when metastore has column rename(but no column reorder), use column index access
there is also a case:
#3 when metastore has column reorder and column rename, there is no way to get it right
This PR only make it configurable to fix #1 and #2, no way to do #3 
What do you think?